frame/elections-phragmen/src/lib.rs

```diff
@@ -334,26 +407,56 @@ decl_module! {
 		///
 		/// A defunct voter is defined to be:
 		///   - a voter whose current submitted votes are all invalid. i.e. all of them are no
-		///     longer a candidate nor an active member.
+		///     longer a candidate nor an active member or a runner-up.
+		///
+		///
+		/// The origin must provide the number of current candidates for the purpose of accurate
+		/// weight calculation.
 		///
 		/// # <weight>
-		/// #### State
-		/// Reads: O(NLogM) given M current candidates and N votes for `target`.
-		/// Writes: O(1)
+		/// No Base weight based on min square analysis.
+		/// State reads:
+		///  	- Voting(reporter)
+		///  	- Candidate.len()
+		///  	- Voting(Target)
+		///  	- Candidates, Members, RunnersUp (is_defunct_voter) [and Voting(Target) -- already counted]
+		/// State writes:
+		/// 	- Lock(reporter || target)
+		/// 	- AccountBalance(reporter) + AccountBalance(target)
+		/// 	- Voting(reporter || target)
+		/// Note: the db access is worse with respect to db, which is when the report is correct.
 		/// # </weight>
-		#[weight = 1_000_000_000]
-		fn report_defunct_voter(origin, target: <T::Lookup as StaticLookup>::Source) {
+		#[weight = FunctionOf(
+			|(defunct,): (&DefunctVoter<<T::Lookup as StaticLookup>::Source>,)|
+				defunct.candidate_count as Weight * (2 * WEIGHT_PER_MICROS) +
+				defunct.vote_count as Weight * (18 * WEIGHT_PER_MICROS) +
+				T::DbWeight::get().reads_writes(6, 4),
+			DispatchClass::Normal,
+			Pays::Yes,
+		)]
+		fn report_defunct_voter(
+			origin,
+			defunct: DefunctVoter<<T::Lookup as StaticLookup>::Source>,
```

Comment:
@shawntabrizi I recall that you once asked about whether if this is needed anymore. this can be swapped with just T::AccountId as well. Not sure if source and lookup is needed here.